.github/workflows/integration-informing.yaml

```diff
@@ -80,9 +80,27 @@ jobs:
       - name: Apply ${{ matrix.patch }} patch
         run: |
           ./build-scripts/patches/${{ matrix.patch }}/apply
+      - name: Generate inspection report
+        run: |
+          sudo k8s/scripts/inspect.sh
+      - name: Upload inspection report artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: inspection-report
+          path: ./inspection-report-*.tar.gz
       - name: Run end to end tests
         run: |
           export TEST_SNAP="$PWD/build/k8s-${{ matrix.patch }}.snap"
           export TEST_SUBSTRATE=lxd
           export TEST_LXD_IMAGE=${{ matrix.os }}
           cd tests/integration && sg lxd -c 'tox -e integration'
+      - name: Generate inspection report
+        if: failure()
+        run: |
+          sudo k8s/scripts/inspect.sh
+      - name: Upload inspection report artifact
+        if: failure()
+        uses: actions/upload-artifact@v4
+        with:
+          name: inspection-report
+          path: ./inspection-report-*.tar.gz
```

Comment:
This needs to run as part of the test harness, as we need the inspection report from the LXC instance(s) (or multipass instances, or the test machines in general).

My recommendation would be: Add a config option `TEST_COLLECT_INSPECTION_REPORTS_DIR`, where we dump inspection reports from each instance on the tests (if not empty). Each inspection report should have the instance name somehow. Then, we would have a step here that `if: failure()`, would upload the directory at the end of the run.

This way, a failed run will give us output that describes what failed (the test output we already have), then we'll also have inspection reports from all instances.

Reply:
This currently runs on the github runner instance, but this is not where k8s-snap is running for our tests

Reply:
will do after I validated that stuff works in `integration.yaml`